pennylane/resource/specs.py

```diff
@@ -272,22 +511,167 @@ def circuit():
+                                    SpecsResources(gate_types={'RandomLayers': 1},
+                                                   gate_sizes={2: 1},
+                                                   measurements={'expval(Prod(num_wires=2, num_terms=2))': 1},
+                                                   num_allocs=3,
+                                                   depth=1)])
+
+    .. details::
+        :title: Using specs on workflows compiled with Catalyst
+
+        The available options for `levels` are different for circuits which have been compiled using Catalyst.
+        There are 2 broad types of specs which can be retrieved for compiled qnodes: runtime resource tracking,
+        and pass-by-pass specs for user applied compilation steps.
+
+        *Runtime resource tracking* (specified by ``level="device"``) works by mock-executing the desired
+        workflow and tracking the number of times a given gate has been applied. This mock-execution happens
+        after all compilation steps, and should be highly accurate to the final gatecounts of running on
+        a real device.
+
+        .. code-block:: python
+
+            qml.capture.enable()  # Enable program capture to allow these transforms to be applied only as MLIR passes
+
+            dev = qml.device("lightning.qubit", wires=3)
+
+            @qml.qjit
+            @qml.transforms.merge_rotations
+            @qml.transforms.cancel_inverses
+            @qml.qnode(dev)
+            def circuit(x):
+                qml.RX(x, wires=0)
+                qml.RX(x, wires=0)
+                qml.X(0)
+                qml.X(0)
+                qml.CNOT([0, 1])
+                return qml.probs()
+
+        ::
+
+            >>> print(qml.specs(circuit, level="device")(1.23))
+            Device: lightning.qubit
+            Device wires: 3
+            Shots: Shots(total=None)
+            Level: device
+            <BLANKLINE>
+            Resource specifications:
+              Total qubit allocations: 3
+              Total gates: 2
+              Circuit depth: 2
+            <BLANKLINE>
+              Gate types:
+                CNOT: 1
+                RX: 1
+            <BLANKLINE>
+              Measurements:
+                No measurements.
+
+        .. warning::
+            Measurement data is not currently supported with runtime resource tracking, so measurement
+            data may show as missing.
+
+        *Pass-by-pass specs* work by analyzing the intermediate representations of compiled circuits.
+        This can be helpful for determining how circuit resources change after a given transform
+        or compiler pass.
+        This version of specs can be applied by passing one of the following values for the `level` argument:
+
+        * An int: the desired pass level of a user-applied pass, see the note below
+        * A marker name (str): The name of an applied :func:`qml.marker <pennylane.marker>` pass
```

Comment:
This doesn't work with program capture, right?

Reply:
When I tested it, markers in general don't seem to work with program capture (at least right now). So it's not an issue unique to specs